app/user.go

```diff
@@ -0,0 +1,597 @@
+package app
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/gogo/protobuf/proto"
+	"github.com/temporalio/tcld/protogen/api/auth/v1"
+	"github.com/temporalio/tcld/protogen/api/authservice/v1"
+	"github.com/urfave/cli/v2"
+	"strings"
+)
+
+const (
+	userIDFlagName              = "user-id"
+	userEmailFlagName           = "user-email"
+	accountRoleFlagName         = "account-role"
+	namespacePermissionFlagName = "namespace-permission"
+)
+
+var (
+	userIDFlag = &cli.StringFlag{
+		Name:    userIDFlagName,
+		Usage:   "The user id",
+		Aliases: []string{"i"},
+	}
+	userEmailFlag = &cli.StringFlag{
+		Name:    userEmailFlagName,
+		Usage:   "The user email address of the user to be invited",
+		Aliases: []string{"e"},
+	}
+)
+
+type UserClient struct {
```

Comment:
do we follow this same pattern for other resource types? Also, why is this exported